crates/autopilot/src/run_loop.rs

```diff
@@ -793,6 +794,18 @@ impl RunLoop {
         auction_id: i64,
         submission_deadline_latest_block: u64,
     ) -> Result<TxId, SettleError> {
+        let current_block = self.eth.current_block().borrow().number;
+        if current_block >= submission_deadline_latest_block {
+            self.in_flight_orders
```

Comment:
It's unfortunate that this duplicates the in-flight order handling. Alternatively you could do the check in a `settle` call like this:
```rust
        let settle = async move {
            let current_block = self.eth.current_block().borrow().number;
            anyhow::ensure!(
                current_block >= submission_deadline_latest_block,
                "submission deadline was missed while waiting for the settlement queue"
            );
            driver.settle(&request, self.config.max_settlement_transaction_wait).await
        };
```
Then you could await `settle.boxed()` in the `select` macro.